docs/gitlab/gitlab-gcp.md

```diff
@@ -0,0 +1,479 @@
+```
+- Using `CLUSTER-REF` get GKECluster resource
+```bash
+kubectl get gkecluster [CLUSTER-REF value] -n crossplane-system
+```
+```
+NAME                                       STATUS   STATE     CLUSTER-NAME                               ENDPOINT          CLUSTER-CLASS          LOCATION        RECLAIM-POLICY   AGE
+gke-af012df6-6e2a-11e9-ac37-9cb6d08bde99   Bound    RUNNING   gke-af11dfb1-6e2a-11e9-ac37-9cb6d08bde99   130.211.208.249   standard-gcp-cluster   us-central1-a   Delete           72m
+```
+- Record the `CLUSTER_NAME` value
+- Obtain GKE Cluster credentials
+    - Note: the easiest way to get `glcoud` command is via:
+        - Go to: https://console.cloud.google.com/kubernetes/list
+        - Click `Connect` next to cluster with `CLUSTER-NAME` value 
+```bash
+gcloud container clusters [CLUSTER-NAME] --zone [CLUSTER-ZONE] --project my-project-123456
+``` 
+
+Add your user account to the cluster admin role
+```bash
+kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole cluster-admin \
+    --user [your-gcp-user-name]
+```
+
+#### External DNS
+- Fetch [Eternal-DNS](https://github.com/helm/charts/tree/master/stable/external-dns) helm chart
+```bash
+helm fetch stable/external-dns
+```
+If `helm fetch` command is successful, you should see a new file created in your CWD:
+```bash
+ls -l external-dns-*
+```
+```
+-rw-r--r-- 1 user user 8913 May  3 23:24 external-dns-1.7.5.tgz
+```
+
+- Render Helm chart into `yaml`, set values and apply to your GKE cluster
+```bash
+helm template external-dns-1.7.5.tgz --name gitlab-demo --namespace kube-system  \
+    --set provider=google \
+    --set txtOwnerId=[gke-cluster-name] \
+    --set google.project=[gcp-project-id] \
+    --set rbac.create=true | kubectl -n kube-system apply -f -
+```
+```
+service/release-name-external-dns created
+deployment.extensions/release-name-external-dns created
+```
+- Verify `Eternal-DNS` is up and running
+```bash
+kubectl get deploy,service -l release=gitlab-demo -n kube-system
+```
+```
+NAME                                             DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+deployment.extensions/gitlab-demo-external-dns   1         1         1            1           1m
+
+NAME                               TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
+service/gitlab-demo-external-dns   ClusterIP   10.75.14.226   <none>        7979/TCP   1m
+```
+
+#### Managed Resource Secrets
+Decide on the GKE cluster namespace where GitLab application artifacts will be deployed.
+
+We will use: `gitlab`, and for convenience we will [set our current context](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#setting-the-namespace-preference) to this namespace 
+```bash
+kubectl create ns gitlab
+kubectl config set-context $(kubectl config current-context) --namespace=gitlab
+```
+
+##### Export and Convert Secrets
```

Comment:
would be great to highlight how we solve this problem going forward, and point an issue.

Reply:
:heavy_check_mark: